task-native/src/main/java/org/ballerinalang/stdlib/task/actions/TaskActions.java

```diff
@@ -37,96 +44,110 @@
  */
 public class TaskActions {
 
-    public static Object pause(BObject taskListener) {
-        String triggerID = (String) taskListener.getNativeData(TaskConstants.TRIGGER_NAME);
-        TaskScheduler taskScheduler = (TaskScheduler) taskListener.getNativeData(TaskConstants.SCHEDULER);
+    private static int bound = 1000000;
+    private static String value = "1000";
+
+    public static Object configureThread(Environment env, long workerCount, long waitingTimeInMillis) {
         try {
-            taskScheduler.pause(triggerID);
-        } catch (SchedulerException e) {
+            TaskManager.getInstance().initializeScheduler(Utils.createSchedulerProperties(
+                    String.valueOf(workerCount), String.valueOf(waitingTimeInMillis)), env);
```

Comment:
Don't we need to check whether there is any running scheduler and restart it with the new configutation

Reply:
If the scheduler not started already, we don't need to start. we store the config and start with the first schedule job

Reply:
I have updated. Can you check [this](https://github.com/ballerina-platform/module-ballerina-task/pull/77/files#diff-c5aa6d18727c94eed961cc14b4c53863a1bf585fdd7d332d1bf43f2a06e94a77R59)